src/domain/session/room/timeline/tiles/TextTile.js

```diff
@@ -16,18 +16,42 @@ limitations under the License.
 
 import {BaseTextTile} from "./BaseTextTile.js";
 import {parsePlainBody} from "../MessageBody.js";
+import {parseHTMLBody} from "../deserialize.js";
 
 export class TextTile extends BaseTextTile {
-    _getBodyAsString() {
+    _getContentString(key, fallback = null) {
         const content = this._getContent();
-        let body = content?.body || "";
+        let val = content?.[key] || fallback;
+        if (!val && val !== "") { // empty string is falsy, but OK here.
+            return null;
+        }
         if (content.msgtype === "m.emote") {
-            body = `* ${this.displayName} ${body}`;
+            val = `* ${this.displayName} ${val}`;
         }
-        return body;
+        return val;
+    }
+
+    _getPlainBody() {
+        return this._getContentString("body", "");
+    }
+
+    _getFormattedBody() {
+        return this._getContentString("formatted_body");
     }
 
-    _parseBody(bodyString) {
-        return parsePlainBody(bodyString);
+    _getBody() {
+        return this._getFormattedBody() || this._getPlainBody();
```

Comment:
can we check the format here rather than the presence of a property in contents?